src/state/opamFormatUpgrade.ml

```diff
@@ -1054,10 +1054,26 @@ let v2_1_alpha2 = OpamVersion.of_string "2.1~alpha2"
 (* config & sw config downgrade opam-version to 2.0 and add opam root version *)
 let v2_1_rc = OpamVersion.of_string "2.1~rc"
 
-let _v2_1 = OpamVersion.of_string "2.1"
+let v2_1 = OpamVersion.of_string "2.1"
 
 let from_2_0_to_2_1_alpha _ conf = conf
 
+let get_unrecorded_local_switch conf =
```

Comment:
I am not sure about this: format upgrade is not expected to depend on CWD ?

The proper way to do it would be to do the update lazily but it's probably not practical...

Another point: it's forbidden to touch an external switch if its opamroot doesn't match the current one, so that must be checked.